### ISSUE TITLE
Unable to complete automatic update to NVDA version next-14685,bc898bdc

### ISSUE THREAD

**Issue by PratikP1** [nvaccess/nvda#7824]
NVDA fails to update to NVDA version next-14685,bc898bdc from version next-14675,14728b95. Testing included updates with and without add-ons disabled. Update gets stuck on progress screen. It has to be manually cancelled. (See log snippet.)

### Steps to reproduce:

1. Go to NVDA's help by pressing nvda+n, h.
2. Choose "check for update"
3. choose to update and bypass the donation screen.
4. Check the following log snippet for details.

> ERROR - gui.installerGui.doInstall (10:51:30.671):
> Failed to execute installer
> Traceback (most recent call last):
>   File "gui\installerGui.pyc", line 32, in doInstall
>   File "config\__init__.pyc", line 188, in execElevated
>   File "winKernel.pyc", line 142, in GetExitCodeProcess
> WindowsError: [Error 6] The handle is invalid.

### Expected behavior:
NVDA should install without a problem.

### Actual behavior:
NVDA fails to update and the installation has to be cancelled.

### System configuration:
next-14675,14728b95. 

NVDA Installed or portable:
Installed 

Windows version:
Windows 10 version 1709 build 17046.100 

### Other questions:

Does the issue still occur after restarting your PC?
Yes.

**Comment by leonardder** [comment]
Have you been able to reproduce this multiple times?

**Comment by PratikP1** [comment]
Yes. I have been able to reproduce this at least five times. I have also just downloaded the snapshot and attempted to update by using that method.

**Comment by Brian1Gaff** [comment]
I had no issues with the portable update here. What version were you 
updating from as a matter of interest?

I use master as my installed copy, so not had to update it as yet as none 
have been done for a few days now.
 Brian

**Comment by Brian1Gaff** [comment]
So does this mean it won't actually download it or is it that it won't run 
the downloaded file?
 Certainly when the portable version downloads via Firefox all seems well 
and so does running of the file from in the downloads dialogue in Firefox. 
That only leaves the previous nvda version as the problem, surely?
 Brian

**Comment by PratikP1** [comment]
It runs but fails during update. The progress remains at 0 percent for more than five minutes.

**Comment by Brian1Gaff** [comment]
Oh well I just did an install of your previous version on windows 7 and did 
an update and had no issues apart from the usual messing about in the log 
during installs of files not being deleted etc. It then booted back in as 
normal.
Speaking [LangChangeCommand ('en_GB'), u'Desktop  list']
IO - speech.speak (19:56:27.913):
Speaking [LangChangeCommand ('en_GB'), u'Computer  2 of 63']
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:27.951):
Python warning:
C:\Users\Brian\AppData\Local\Temp\nsl993.tmp\app\library.zip\configobj.py:28: 
DeprecationWarning: The compiler package is deprecated and removed in Python 
3.x.
IO - speech.speak (19:56:28.025):
Speaking [LangChangeCommand ('en_GB'), u'Updating NVDA  dialog  Please wait 
while your previous installation of NVDA is being updated.']
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:28.036):
installer.tryCopyFile:
Unable to copy 
\\?\C:\Users\Brian\AppData\Local\Temp\nsl993.tmp\app\nvda_eoaProxy.exe, 
error 32
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:28.040):
installer.tryCopyFile:
Unable to copy 
\\?\C:\Users\Brian\AppData\Local\Temp\nsl993.tmp\app\python27.dll, error 32
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:28.367):
installer.tryCopyFile:
Unable to copy 
\\?\C:\Users\Brian\AppData\Local\Temp\nsl993.tmp\app\_ctypes.pyd, error 32
IO - tones.beep (19:56:28.875):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
IO - speech.speak (19:56:28.877):
Speaking [LangChangeCommand ('en_GB'), u'Please wait']
IO - tones.beep (19:56:28.907):
Beep at pitch 110.0, for 40 ms, left volume 50, right volume 50
IO - speech.speak (19:56:28.911):
Speaking [LangChangeCommand ('en_GB'), u'0 percent']
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:29.517):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\IAccessible2proxy.dll'
IO - tones.beep (19:56:29.888):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:30.519):
installer.tryRemoveFile:
Failed to delete file C:\Program Files 
(x86)\NVDA\lib\next-14675,14728b95\tmpgewxzy, marking for delete on reboot
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:30.523):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\ISimpleDOM.dll'
IO - tones.beep (19:56:30.904):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:31.526):
installer.tryRemoveFile:
Failed to delete file C:\Program Files 
(x86)\NVDA\lib\next-14675,14728b95\tmpgttfnl, marking for delete on reboot
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:31.532):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\minHook.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:31.535):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\nvdaHelperLocal.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:31.538):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\nvdaHelperLocalWin10.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:31.540):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\nvdaHelperRemote.dll'
IO - inputCore.InputManager.executeGesture (19:56:31.740):
Input: kb(desktop):alt+tab
IO - tones.beep (19:56:31.920):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:32.543):
installer.tryRemoveFile:
Failed to delete file C:\Program Files 
(x86)\NVDA\lib\next-14675,14728b95\tmpnr9hzx, marking for delete on reboot
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.546):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\VBufBackend_adobeAcrobat.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.549):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\VBufBackend_adobeFlash.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.552):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\VBufBackend_gecko_ia2.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.555):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\VBufBackend_lotusNotesRichText.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.559):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\VBufBackend_mshtml.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.563):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95\\VBufBackend_webKit.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.568):
installer.removeOldLibFiles:
Skipping current install lib path: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14685,bc898bdc'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.569):
installer.removeOldLibFiles:
Removing old lib directory: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14675,14728b95'
WARNING - RPC process 3596 (nvda_slave.exe) (19:56:32.572):
installer.removeOldLibFiles:
Failed to remove a directory no longer needed. This can be manually removed 
after a reboot or the  installer will try removing it again next time. 
Directory: u'C:\\Program Files (x86)\\NVDA\\lib\\next-14675,14728b95'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.575):
installer.removeOldLibFiles:
Removing old lib directory: u'C:\\Program Files 
(x86)\\NVDA\\lib\\next-14685,bc898bdc'
WARNING - RPC process 3596 (nvda_slave.exe) (19:56:32.576):
installer.removeOldLibFiles:
Failed to remove a directory no longer needed. This can be manually removed 
after a reboot or the  installer will try removing it again next time. 
Directory: u'C:\\Program Files (x86)\\NVDA\\lib\\next-14685,bc898bdc'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:32.582):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\IAccessible2proxy.dll'
IO - tones.beep (19:56:32.930):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:33.585):
installer.tryRemoveFile:
Failed to delete file C:\Program Files 
(x86)\NVDA\lib64\next-14675,14728b95\tmplofbt0, marking for delete on reboot
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:33.588):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\ISimpleDOM.dll'
IO - tones.beep (19:56:33.944):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - RPC process 3596 (nvda_slave.exe) (19:56:34.591):
installer.tryRemoveFile:
Failed to delete file C:\Program Files 
(x86)\NVDA\lib64\next-14675,14728b95\tmpvtyzzh, marking for delete on reboot
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.594):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\minHook.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.596):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\nvdaHelperRemote.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.599):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\nvdaHelperRemoteLoader.exe'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.604):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\VBufBackend_adobeAcrobat.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.608):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\VBufBackend_adobeFlash.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.611):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\VBufBackend_gecko_ia2.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.614):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\VBufBackend_lotusNotesRichText.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.617):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\VBufBackend_mshtml.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.619):
installer.removeOldLibFiles:
Removing old lib file: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95\\VBufBackend_webKit.dll'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.624):
installer.removeOldLibFiles:
Skipping current install lib path: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14685,bc898bdc'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.625):
installer.removeOldLibFiles:
Removing old lib directory: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\master-14676,f2bf8884'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.627):
installer.removeOldLibFiles:
Removing old lib directory: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14675,14728b95'
WARNING - RPC process 3596 (nvda_slave.exe) (19:56:34.628):
installer.removeOldLibFiles:
Failed to remove a directory no longer needed. This can be manually removed 
after a reboot or the  installer will try removing it again next time. 
Directory: u'C:\\Program Files (x86)\\NVDA\\lib64\\next-14675,14728b95'
DEBUG - RPC process 3596 (nvda_slave.exe) (19:56:34.628):
installer.removeOldLibFiles:
Removing old lib directory: u'C:\\Program Files 
(x86)\\NVDA\\lib64\\next-14685,bc898bdc'
WARNING - RPC process 3596 (nvda_slave.exe) (19:56:34.631):
installer.removeOldLibFiles:
Failed to remove a directory no longer needed. This can be manually removed 
after a reboot or the  installer will try removing it again next time. 
Directory: u'C:\\Program Files (x86)\\NVDA\\lib64\\next-14685,bc898bdc'
IO - tones.beep (19:56:34.640):
Beep at pitch 1760, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole 
(19:56:34.776):
accRole failed: (-2147024809, 'The parameter is incorrect.', (None, None, 
None, 0, None))
IO - speech.speak (19:56:34.802):
Speaking [LangChangeCommand ('en_GB'), u'Desktop  list']
IO - speech.speak (19:56:34.821):
Speaking [LangChangeCommand ('en_GB'), u'Computer  2 of 63']
IO - inputCore.InputManager.executeGesture (19:56:34.931):
Input: kb(desktop):alt+tab
DEBUGWARNING - RPC process 4836 (nvda_update_7wz51m.exe) (19:56:35.924):
Thread 5028, build\x86\remote\ia2Support.cpp, 
IA2Support_inProcess_terminate, 152:
WaitForMultipleObjects returned 258

Those who know me know I feel the sudden ending of the log in this way is 
why sometimes my updates stall, but this time it made it and rebooted OK.
 No sign of the error you got here. Must be either some other software, 
something to do with Windows 10 or its settings or some weird corruption.
 Brian

**Comment by PratikP1** [comment]
This might be related to Windows 10. I can’t tell. My log repeatedly shows the output I included in my report.

**Comment by Brian1Gaff** [comment]
The only funny thing was that updating from a master to a next said I was 
attempting to use an older version, and yet now I've just put things back as 
they were the same message appeared and the reboot crashed and i had to 
restart nvda using narrator.
 I do not think this is related, but its curious as I'd have thought next 
was newer than master.

Anyway, have you got a different anti virus, security settings or some other 
type of program which might be able to interfere at that level?

I'm assuming if you alt tab as I did in my install you saw, that there was 
no error hidden behind the  main window? This happened to me once and it was 
a UAC window and because the updater was running you could not read what it 
was saying.


Brian

**Comment by PratikP1** [comment]
No. No AV or other such issues. I tried this multiple times including after a fresh restart. I don’t believe in third-party AV.

**Comment by feerrenrut** [comment]
@PratikP1 I tried updating between the versions specified on a Windows 10 machine and was unable to reproduce the error you are getting. I tried this with UAC settings at 34%, 66%, and 100%.

To ensure this is not related to any add-on or other config, if you are comfortable to do so, please remove (and backup) your nvda config directory. Then try to update NVDA again.

To do this:
1. Open the windows run dialog with `windows+r`
1. Type `%appdata%` and press enter
1. Locate the `nvda` folder and rename (`F2`) to `nvda_backup`
1. Try to update nvda again.

Please report back on how this goes.

To restore your settings:
1. Open the windows run dialog with `windows+r`
1. Type `%appdata%` and press enter
1. Ensure there is a directory called `nvda_backup`.  If not, stop here.
1. If there is a directory called `nvda`, delete it.
1. Locate the `nvda_backup` folder and rename (`F2`) to `nvda`

**Comment by PratikP1** [comment]
I was unable to update after renaming the NVDA config Directory. After leaving it for a while and restarting, I was able to update to 14691,a54afde0 without a problem.

**Comment by Brian1Gaff** [comment]
That is most peculiar.
 I do not see any real need to alter this as its surely just as it was 
before??

Brian

**Comment by feerrenrut** [comment]
Hmm, strange. While I'm not exactly happy to leave it like this, without some new avenue to investigate to reproduce this issue there isn't much we can do. I'll close this issue, if something new comes up please reopen / comment.

**Comment by Brian1Gaff** [comment]
OK got it, I think.
 One issue seems to be that if you have anything that needs to remove the 
system config folder, sometimes it fails with a hard error and just goes 
around in circles.
 I found this with my current master copy when I did some updates to add ons 
then went to general settings and tried to hit the button to copy the 
settings so they would be used at log in. it just bleeped for ever, on doing 
an alt tab I found a retry and cancell error hidden behind the updating 
page.

Here is a log fragment of this bit.

IO - inputCore.InputManager.executeGesture (07:28:38.868):
Input: kb(desktop):tab
IO - speech.speak (07:28:38.904):
Speaking [LangChangeCommand ('en_GB'), u'Cancel  button']
IO - inputCore.InputManager.executeGesture (07:28:40.625):
Input: kb(desktop):shift+tab
IO - speech.speak (07:28:40.663):
Speaking [LangChangeCommand ('en_GB'), u'Retry  button  Alt+r']
IO - inputCore.InputManager.executeGesture (07:28:42.131):
Input: kb(desktop):space
IO - speech._speakSpellingGen (07:28:42.135):
Speaking character u'space'
DEBUG - queueHandler.registerGeneratorObject (07:28:42.135):
Adding generator 3276
IO - speech.speak (07:28:42.137):
Speaking [LangChangeCommand ('en_GB'), u'pressed']
DEBUG - queueHandler.pumpAll (07:28:42.154):
generator 3276 finished
DEBUGWARNING - gui.ExecAndPump.run (07:28:42.240):
task had errors
Traceback (most recent call last):
  File "gui\__init__.pyc", line 824, in run
  File "config\__init__.pyc", line 219, in setSystemConfigToCurrentConfig
  File "config\__init__.pyc", line 231, in _setSystemConfig
  File "installer.pyc", line 348, in tryRemoveFile
RetriableFailure: Failed to rename file C:\Program Files 
(x86)\NVDA\systemConfig before  remove
DEBUGWARNING - gui.settingsDialogs.GeneralSettingsDialog.onCopySettings 
(07:28:42.344):
Error when copying settings to system config
Traceback (most recent call last):
  File "gui\settingsDialogs.pyc", line 235, in onCopySettings
  File "gui\__init__.pyc", line 820, in __init__
RetriableFailure: Failed to rename file C:\Program Files 
(x86)\NVDA\systemConfig before  remove
IO - speech.speak (07:28:42.387):
Speaking [LangChangeCommand ('en_GB'), u'Error Copying  dialog  Unable to 
copy a file. Perhaps it is currently being used by another process or you 
have run out of disc space on the drive you are copying to.']
IO - speech.speak (07:28:42.391):
Speaking [LangChangeCommand ('en_GB'), u'Retry  button  Alt+r']

Cancel gets you out then it says it could not copy the settings. However It 
could under some circumstances, presumably, need to do this process when 
updating, and then the same problem would happen one assumes.
 Brian

**Comment by Brian1Gaff** [comment]
Just did, but not sure if its the same issue or another one really.

Brian

**Comment by Brian1Gaff** [comment]
I did exactly what the  reporter did, went off after the error and then came 
back and tried it again and of course it worked this time.

 Could this be linked to the same issue we had on updates with the folder 
deletes before, and Mick somehow made them not cause the hard error?

Brian

**Comment by feerrenrut** [comment]
@Brian1Gaff This looks like a different problem. I will move this to a new issue, to keep this issue clear I will remove the comments from here relating to the "locked" config files

**Comment by feerrenrut** [comment]
@Brian1Gaff Please see #7830

**Comment by Brian1Gaff** [comment]
OK. All I did was go into the prefs, general and hit the button to allow the 
current settings to be used on the log on screen.

 In the case of the error for not restarting that is just simply updating a 
copy of nvda installed or portable and sometimes it fails to restart 
complaining about the log file  on an error box and cancelling this allows a 
manual restart to occur. At this time a log is written to the  main nvda 
folder called nvda.exe.log, detailing the inability to use the log file as 
in another issue.
 I am not sure if all these are interrelated. they do share some aspects in 
that files are locked or unable to be written to.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by Brian1Gaff** [comment]
I'm thinking this might be due to something in Windows hanging on to folders 
or files when its not supposed to. Other than that I'm baffled by all of 
these strange goings on.

**Comment by PratikP1** [comment]
@feerrenrut I'm afraid this issue has occurred once again. I was updating from version next-14691,a54afde0 to version next-14694,1a66633b. I had the same pattern: The installer got stuck and remained beeping for minutes before me needing to cancel the installation.
I have UAC completely switched off. With careful management, I find that I do not need it.

**Comment by michaelDCurran** [comment]
It may be in fact UAC being switched off which is part of the problem.


Note that files are created differently with UAC on verses off. I.e. an 
installer that is first used with UAC  will create files perhaps  
needing admin rights. Where as then running the installer with no UAC, 
it does not elevate and therefore cannot use admin rights.


Can you try installing again with your log set to debug and provide the 
output just before you cancel it?


If this yields no answers, we'll have to start adding more debug log 
output and get you to try that.

**Comment by PratikP1** [comment]
@michaelDCurran commented 
> It may be in fact UAC being switched off which is part of the problem.
This problem just started. Unless something has happened recently, I'm not sure how it could be related. It updates fine after several restarts. I haven't succeeded this time around yet. Could it be related to #7002 825?

**Comment by feerrenrut** [comment]
I also could not reproduce this issue with UAC at level 0.
@PratikP1 Could you please attach a full debug level log to this issue?

We have identified some extra information we could include in the log to track down this issue, however it will help us to focus our efforts if we could first inspect your log file.

**Comment by Brian1Gaff** [comment]
Is that really the case. I never knew that. I got so fed up with it moaning 
I just threw it off completely myself.

It seems a bit pointless having it if its going to not realise you now have 
it off.
 OK I use 7, but I bet the underlying code is the same. Brian

**Comment by Brian1Gaff** [comment]
I was never quite sure if moving it to zero really meant zero, so.

Windows Registry Editor Version 5.00


[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System]
"EnableLUA"=dword:00000001

Change the last 1 to a 0 and its off

Note this works in Windows 7, no idea about 10!
.
 Brian

**Comment by PratikP1** [comment]
@feerrenrut commented 
> Could you please attach a full debug level log to this issue? We have identified some extra information we could include in the log to track down this issue, however it will help us to focus our efforts if we could first inspect your log file.

I got it to install after restarting several times again. Unfortunately the log was overridden. I should have included it the last time around. My apologies. The error I listed was the only suspicious thing in the log. That happened after I interrupted the process. otherwise, the only message I recall seeing was the progress beeps.

I have never had to switch to different logging levels during installation and have never had to look at the general configuration settings. However you should know that navigating in the general dialog by using tab and shift+tab is not possible. Object navigation needs to be used to change logging levels. Focusing on the combo box, opening it, and selecting the correct option also requires object navigation. This occurred during the failed installs. I can't tell whether the lack of navigation is by design, accident, or a bug. Nor can I tell whether this behavior is an artifact of the current problem. It was the next thing for me to report in a separate issue. Is there a way to pause installation while I configure logging levels? It takes far longer to achieve it with object navigation. You may not get every relevant details. There should be a configurable gesture/command that allows us to cycle quickly through logging levels. But that's for another issue.

I suspect this installation issue hasn't gone away. I will let you know when a new build arrives.  I'll do my best to get you the log.

**Comment by PratikP1** [comment]
@feerrenrut I'm attaching a full debug log as requested. I'm highlighting a few of my actions below in order to annotate and reduce the time that you have to look through seemingly odd entries.
* This time the update was to NVDA version next-14699,4f6fe6f3
* beeps start at this time location: IO - tones.beep (23:59:58.832):
* beeps interrupted for me to check logging level set to Debug. I had already set the level before starting update and saved configuration. DEBUG - synthDrivers.oneCore.SynthDriver.cancel (00:00:01.910): Cancelling
* IO - inputCore.InputManager.executeGesture (00:00:01.993):
* note the nonfunctional tab key in this dialog, thus the usage of object navigation
* Returning to the regularly scheduled beeps: IO - speech.speak (00:00:11.184)
* I notice that I've been sitting on my hands for five minutes. Beeps continue until I open log viewer and copy since the last attempt at an upgrade at 11:45 resulted in NVDA old log not being saved. IO - inputCore.InputManager.executeGesture (00:05:58.594)
* I initiate cancellation of installation. Core.InputManager.executeGesture (00:06:21.816)
* I used ctrl+alt+n rather than exit as previous attempts to use escape or nvda+q resulted in no nvda old logs being preserved.

[nvda-old-log.txt](https://github.com/nvaccess/nvda/files/1537966/nvda-old-log.txt)

**Comment by michaelDCurran** [comment]
Note I've currently got my machine in a state that is reproducing this. 
I'm not rebooting :)


It looks like the way we elevate (using ShellExecuteEx with a verb of 
"runAs") is not highly recommended by Microsoft. Rather they would 
rather we correctly set requireAdministrator in the executable's manifest.


What we have been doing in the past has worked clearly until recently.


Something must have changed in Windows 10.


@PratikP1 are you on Windows 10 Fall Creators update and or a recent 
insider build?

**Comment by PratikP1** [comment]
@michaelDCurran commented
> are you on Windows 10 Fall Creators update and or a recent insider build?

I'm on an insider build:  Windows 10 1709 build 17046.100. Even within builds, there could be different code which might affect what is going on here--A/B testing.

**Comment by Brian1Gaff** [comment]
In windows 7 when I got that issue of non restart after an update it too 
failed to have the old log renamed and generated the  log in the program 
folder about the file not being accessible.
 Just in case these are linked issues.
